cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModelStats.java

```diff
@@ -363,4 +384,33 @@ private void numForAvgTopicReplicas(ClusterModel clusterModel) {
     _topicReplicaStats.put(Statistic.AVG, _topicReplicaStats.get(Statistic.AVG).doubleValue() / _numTopics);
     _topicReplicaStats.put(Statistic.ST_DEV, _topicReplicaStats.get(Statistic.ST_DEV).doubleValue() / _numTopics);
   }
+
+  /**
+   * Generate statistics for disks in the given cluster.
+   *
+   * @param clusterModel The state of the cluster.
+   * @param balancingConstraint Balancing constraint.
+   */
+  private void numForDisks(ClusterModel clusterModel, BalancingConstraint balancingConstraint) {
+    double totalDiskUtilizationVariance = 0;
+    int diskCount = 0;
+    for (Broker broker : clusterModel.aliveBrokers()) {
+      double brokerDiskUtilization = averageDiskUtilizationPercentage(broker);
+      double upperLimit = brokerDiskUtilization * balancingConstraint.resourceBalancePercentage(Resource.DISK);
+      double lowerLimit = brokerDiskUtilization * Math.max(0, (2 - balancingConstraint.resourceBalancePercentage(Resource.DISK)));
+      for (Disk disk : broker.disk()) {
+        if (disk.state() == Disk.State.DEAD) {
+          continue;
+        }
+        if (diskUtilizationPercentage(disk) > upperLimit || diskUtilizationPercentage(disk) < lowerLimit) {
+          _numUnbalancedDisk++;
+        }
+        totalDiskUtilizationVariance += Math.pow(diskUtilizationPercentage(disk) - brokerDiskUtilization, 2);
+        diskCount++;
+      }
+    }
+    if (diskCount > 0) {
```

Comment:
This being `0` means `BrokerCapacityResolver` does not return disk capacity information or all the disk is dead(the whole cluster is dead). In this case we cannot get statistics about disk utilization.

The disk does not necessary to be JBOD disk, it can be RAID disk.